source/common/config/config_provider_impl.h

```diff
@@ -49,27 +50,36 @@ namespace Config {
 //   1) Create a class derived from ConfigProviderManagerImplBase and implement the required
 //   interface.
 //      When implementing createXdsConfigProvider(), it is expected that getSubscription<T>() will
-//      be called to fetch either an existing ConfigSubscriptionInstanceBase if the config source
-//      configuration matches, or a newly instantiated subscription otherwise.
+//      be called to fetch either an existing ConfigSubscriptionCommonBase if the config
+//      source configuration matches, or a newly instantiated subscription otherwise.
 //
 // For immutable providers:
-//   1) Create a class derived from ImmutableConfigProviderImplBase and implement the required
+//   1) Create a class derived from ImmutableConfigProviderBase and implement the required
 //   interface.
 //
 // For mutable (xDS) providers:
-//   1) Create a class derived from MutableConfigProviderImplBase and implement the required
-//   interface.
-//   2) Create a class derived from ConfigSubscriptionInstanceBase; this is the entity
-//   responsible for owning and managing the Envoy::Config::Subscription<ConfigProto> that provides
-//   the underlying config subscription.
+//   1) According to the API type, create a class derived from MutableConfigProviderBase or
+//   DeltaMutableConfigProviderBase and implement the required interface.
+//   2) According to the API type, create a class derived from ConfigSubscriptionInstance or
+//   DeltaConfigSubscriptionInstance; this is the entity responsible for owning and managing the
+//   Envoy::Config::Subscription<ConfigProto> that provides the underlying config subscription.
+//     a) For a ConfigProvider::ApiType::Full subscription instance (i.e., a
+//     ConfigSubscriptionInstance child):
 //     - When subscription callbacks (onConfigUpdate, onConfigUpdateFailed) are issued by the
-//     underlying subscription, the corresponding ConfigSubscriptionInstanceBase functions must be
-//     called as well.
-//     - On a successful config update, checkAndApplyConfig() should be called to instantiate the
-//     new config implementation and propagate it to the shared config providers and all
-//     worker threads.
-//       - On a successful return from checkAndApplyConfig(), the config proto must be latched into
-//       this class and returned via the getConfigProto() override.
+//     underlying subscription, the corresponding ConfigSubscriptionInstance functions
+//     must be called as well.
+//     - On a successful config update, checkAndApplyConfigUpdate() should be called to instantiate
```

Comment:
It sounds like the real difference between checkAndApplyConfigUpdate and applyConfigUpdate is that one is for delta, and the other is for state-of-the-world. Is that the case? If so, it would be clearer if their names reflected that.

Reply:
Yeah, that's the case. I can use a more descriptive name to make their behavior explicit as opposed to implicit (based on the object's type).

Reply:
Done.

Note that I only renamed `applyConfigUpdate()` -> `applyDeltaConfigUpdate()` to maintain consistency with the naming convention that non delta types simply avoid the "Full/Delta" qualifier.